pandas/tests/groupby/test_groupby.py

```diff
@@ -1716,15 +1716,40 @@ def test_pivot_table_values_key_error():
         )
 
 
-def test_empty_dataframe_groupby():
-    # GH8093
+@pytest.mark.parametrize("columns", ["C", ["C"]])
+@pytest.mark.parametrize("keys", [["A"], ["A", "B"]])
+@pytest.mark.parametrize(
+    "dtypes",
+    [
+        "object",
+        "int",
+        "float",
+        {"A": "object", "B": "int", "C": "float"},
+        {"A": "int", "B": "float", "C": "object"},
+    ],
```

Comment:
can you add some other datatypes to make sure preserving (categorical, datetime, datetime w/tz, Int). if some still don't work, just xfail them and create an issue.)